server/auth/types/authentication_type.ts

```diff
@@ -73,10 +73,7 @@ export interface OpenSearchDashboardsAuthState {
 }
 
 export abstract class AuthenticationType implements IAuthenticationType {
-  protected static readonly ROUTES_TO_IGNORE: string[] = [
-    '/api/core/capabilities', // FIXME: need to figure out how to bypass this API call
-    '/app/login',
-  ];
+  protected static readonly ROUTES_TO_IGNORE: string[] = ['/app/login'];
```

Comment:
What about we do it in this way:
Add a new constant:
```ts
protected static readonly ROUTES_AUTH_OPTIONAL: string[] = ['/api/core/capabilities'];
```

Then we can have a more general function
```ts
  authOptional(request: OpenSearchDashboardsRequest): boolean {
    const pathname = request.url.pathname;
    if (!pathname) {
      return false;
    }
    // allow requests to ignored routes
    if (AuthenticationType.ROUTES_AUTH_OPTIONAL.includes(pathname!)) {
      return true;
    }
    return false;
  }
```

Finally, instead of calling `this.isCapabilitiesRequest`, we call `this.authOptional`.

what do you think?

Reply:
+1 we should not be ignoring /app/login as a whole. Instead we should narrow it down to capabilities requests and only skip auth for those.

Reply:
Thanks, the code has been updated.